src/hotspot/share/logging/logOutput.cpp

```diff
@@ -339,3 +339,41 @@ void LogOutput::update_config_string(const size_t on_level[LogLevel::Count]) {
   FREE_C_HEAP_ARRAY(Selection, selections);
 }
 
+bool LogOutput::parse_options(const char* options, outputStream* errstream) {
+  if (options == NULL || strlen(options) == 0) {
+    return true;
+  }
+  bool success = true;
+  char* opts = os::strdup_check_oom(options, mtLogging);
+
+  char* comma_pos;
+  char* pos = opts;
+  do {
+    comma_pos = strchr(pos, ',');
+    if (comma_pos != NULL) {
+      *comma_pos = '\0';
+    }
+    char* equals_pos = strchr(pos, '=');
+    if (equals_pos == NULL) {
+      errstream->print_cr("Invalid option '%s' for log output (%s).", pos, name());
+      success = false;
+      break;
+    }
+
+    char* key = pos;
+    char* value_str = equals_pos + 1;
+    *equals_pos = '\0';
+    julong errstream_count_before = errstream->count();
+    success = set_option(key, value_str, errstream);
+    if (!success) {
+      if (errstream->count() > errstream_count_before) {
+        errstream->print_cr("Invalid option '%s' for log output (%s).", key, name());
```

Comment:
I don't quite follow this. If the error count has increased then this was a known option with a bad value that has already been reported in set_option. Here we should be catching an unknown option that wasn't recognized by set_option, in which case the error count should be unchanged no?

Reply:
Good catch! I fixed it in new commit.